hasher-matcher-actioner/hmalib/lambdas/pdq/pdq_matcher.py

```diff
@@ -88,11 +112,19 @@ def lambda_handler(event, context):
                 ).write_to_table(records_table)
 
                 match_ids.append(signal_id)
-            sns_client.publish(
-                TopicArn=OUTPUT_TOPIC_ARN,
-                Subject="Match found in pdq_matcher lambda",
-                Message=f"Match found for key: {key}, hash: {hash_str}, for IDs: {match_ids}",
-            )
+                message = MatchMessage(
+                    content_key=key,
+                    content_hash=hash_str,
+                    banked_indicator_id=signal_id
+                )
+
+                # Publish one message per match. This might be a perf penalty if
+                # we get > 1 match per index query. But I'm assuming that's
+                # rare. Unfortunately, there is no batch publish in SNS.
+                sns_client.publish(
+                    TopicArn=OUTPUT_TOPIC_ARN,
+                    Message=message.to_sns_message()
+                )
```

Comment:
(Question): Which component will handle combining actions from multiple sources and taking only the highest priority one? I.e. if one action is delete the content, and another is send the content for review, the delete should win

Reply:
In my head, that is part of the decisions component. Which is TBH, not making it to V0.

When decisions component exists, the actions layer will become a labeler. It will recommend certain actions based on the match data and metadata. A decisions component will receive all labels and decide which action is 'applicable'.